addons/udes_security/controllers/main.py

```diff
@@ -0,0 +1,135 @@
+            "/web/content/<int:id>-<string:unique>",
+            "/web/content/<int:id>-<string:unique>/<string:filename>",
+            "/web/content/<string:model>/<int:id>/<string:field>",
+            "/web/content/<string:model>/<int:id>/<string:field>/<string:filename>",
+        ],
+        type="http",
+        auth="public",
+    )
+    def content_common(
+        self,
+        xmlid=None,
+        model="ir.attachment",
+        id=None,
+        field="datas",
+        filename=None,
+        filename_field="datas_fname",
+        unique=None,
+        mimetype=None,
+        download=None,
+        data=None,
+        token=None,
+        access_token=None,
+        **kw
+    ):
+        BlockedFileType = request.env["udes.blocked_file_type"].sudo()
+
+        if request.session.uid != SUPERUSER_ID:
+            # Check if the file type is blocked
+            if download:
+                file_type = ""
+                record_filename = ""
+
+                if filename:
+                    file_type = self._get_file_type(filename)
+                elif id and model and filename_field:
+                    RecordModel = request.env[model].sudo()
+                    record = RecordModel.browse(id)
+                    record_filename = record[filename_field]
+                    file_type = self._get_file_type(record_filename)
+
+                blocked_file_type_count = BlockedFileType.search_count([("name", "=", file_type)])
+
+                # If the user is trying to download a blocked file type then 
+                # prevent download and return an error message
+                if blocked_file_type_count:
+                    _logger.info(
+                        "Blocked attempt to download file %s" % (filename or record_filename)
+                    )
+
+                    download_error = {
+                        "message": "Unable to download file: File type blocked",
+                        "data": {
+                            "debug": "File type '%s' has been blocked by system administrator."
+                            % (file_type),
+                        },
+                    }
+
+                    return request.make_response((json.dumps(download_error)))
+
+        return super(BinaryExtension, self).content_common(
+            xmlid,
+            model,
+            id,
+            field,
+            filename,
+            filename_field,
+            unique,
+            mimetype,
+            download,
+            data,
+            token,
+            access_token,
+            **kw
+        )
+
+    @http.route("/web/binary/upload_attachment", type="http", auth="user")
+    @serialize_exception
+    def upload_attachment(self, callback, model, id, ufile):
+
+        BlockedFileType = request.env["udes.blocked_file_type"].sudo()
+
+        files = request.httprequest.files.getlist("ufile")
+
+        if request.session.uid != SUPERUSER_ID or True:
+            for ufile in files:
```

Comment:
This `ufile` is shadowing the function parameter, dangerous if we ever get to line 133.

Reply:
Yeah, it's how Odoo do it in the upload_attachment function in web module hence why I copied that approach. But it is unnecessarily dangerous so will go rename that.

Reply:
Have now changed ```ufile``` to ```upload_file``` to avoid this.